packages/broker/src/storage/StorageConfig.ts

```diff
@@ -144,8 +145,20 @@ export class StorageConfig {
                     this._removeStreams(keys)
                 }
             }
-        })
+        }
+        networkNode.addMessageListener(messageListener)
         networkNode.subscribe(assignmentStreamId, 0)
+        return messageListener
+    }
+
+    stopAssignmentEventListener(messageListener: (msg: Protocol.StreamMessage) => void, streamrAddress: string, networkNode: NetworkNode) {
+        networkNode.removeMessageListener(messageListener)
+        const assignmentStreamId = this.getAssignmentStreamId(streamrAddress)
+        networkNode.unsubscribe(assignmentStreamId, 0)
```

Comment:
Wondering actually if `subscriptionManager` should be used in this class? To guard from unsubscribing from streams that other plugins have subscribed to?

Reply:
Now it uses `subscriptionManager`